scripts/misc/archive_landsat_orphans.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+set -e
+
+<<'###'
+Read landsat orphan report for Landsat 5, Landsat 7 and Landsat 8 and archive datasets
+
+pre-req:
+- read/write/delete access to `deafrica-landsat` and `deafrica-services` s3 buckets
+- admin access to odc database
+
+steps:
+- Read latest orphan report
+- For each orphan path, start archiving datasets. Make sure to archive wofs and fc derived datasets and generate report:
+    * Get dataset-ids using s3-uri:
+      datacube dataset uri-search <s3-uri>
+    * For each dataset-ids:
+      * Get dataset info for given dataset-id
+        datacube dataset info --show-derived <dataset-id> | yq '.'
+      * Generate archived report for orphan and derived datasets: dataset-id, product and location
+      * archive orphan dataset and derived datasets
+        datacube dataset archive --dry-run <derived-id>
+        datacube dataset archive --dry-run <dataset-id>
+- Publish archived report to s3 - s3://deafrica-landsat/status-report/archived/
+###
+
+echo "start archiving Landsat orphan datasets"
+
+if [[ -z ${DB_ADMIN_PASSWORD} || -z ${DB_HOSTNAME} ]]; then
+  echo "Please provide following env variables: DB_HOSTNAME and DB_ADMIN_PASSWORD"
+  exit 1;
+fi
+
+export DB_USERNAME=${DB_ADMIN_USER:-"odc_admin"}
+export DB_DATABASE=${DB_DATABASE:-"odc"}
+export DB_PORT=${DB_PORT:-"5432"}
+export DB_PASSWORD=$DB_ADMIN_PASSWORD
+export DB_HOSTNAME=$DB_HOSTNAME
+
+# Verify db connection
+datacube system check
+
+# 1. Read Report
+if [ -z LATEST_ORPHAN_REPORT ]; then
+  ORPHAN_REPORT_S3_PATH="s3://deafrica-landsat/status-report/orphan/"
+  LATEST_ORPHAN_REPORT=`aws s3 ls $ORPHAN_REPORT_S3_PATH | grep "landsat_orphan_" | sort | tail -n 1 | awk '{print $4}'`
```

Comment:
It is preferred to rather enclose commands in `$()` rather than back ticks, it makes it slightly easier to read